wgpu-core/src/device/global.rs

```diff
@@ -2102,6 +2102,11 @@ impl Global {
         }
     }
 
+    pub fn device_unregister_device_lost_closure(&self, device_id: DeviceId) {
+        let device = self.hub.devices.get(device_id);
+        device.device_lost_closure.lock().take();
```

Comment:
This causes a panic when dropped, because the `consumed` member in each closure variant can still be `false`.

Pushed 6dcdb630f75bf816ebd5b6a1a21588f3ceca6054 as a `fixup!` to disarm it by `call`ing the closure with the `ReplacedCallback` reason for now. I question the design of having these closures guaranteed to be called, but that's obviously the design, so I'll maintain it for now.